master/buildbot/steps/source/git.py

```diff
@@ -222,14 +217,26 @@ def mode_incremental(self, _):
         yield self._syncSubmodule(None)
         yield self._updateSubmodule(None)
 
+    @defer.inlineCallbacks
     def clean(self):
         command = ['clean', '-f', '-f', '-d']
-        d = self._dovccmd(command)
-        d.addCallback(self._fetchOrFallback)
-        d.addCallback(self._syncSubmodule)
-        d.addCallback(self._updateSubmodule)
-        d.addCallback(self._cleanSubmodule)
-        return d
+        rc = yield self._dovccmd(command)
+        if rc != RC_SUCCESS:
+            raise buildstep.BuildStepFailed
+
+        rc = yield self._fetchOrFallback()
+        if rc != RC_SUCCESS:
+            raise buildstep.BuildStepFailed
```

Comment:
What about refactor in a loop:

for clean_meth in ['_fetchOrFallback', '_syncSubmodule', ...]:
    rc = yield getattr(self, clean_meth)
    if rc != RC_SUCCESS:
        raise buildstep.BuildStepFailed